microcosm_pubsub/codecs.py

```diff
@@ -75,21 +80,19 @@ def encode(self, dct=None, **kwargs):
         """
         message = dct.copy() if dct else dict()
         message.update(kwargs)
-        encoded = self.schema.load(message)
-        return dumps(encoded.data)
+        return dumps(self.schema.load(message, unknown=EXCLUDE))
```

Comment:
This is to accommodate during `produce` where we pass in `uri` unconditionally, which otherwise fails the `IdentityMessage` validation.

Reply:
Please add a comment